### ISSUE TITLE
Using Strawberryshake client in Unity

### ISSUE THREAD

**Issue by GMyjak** [ChilliCream/hotchocolate#4066]
**Is your feature request related to a problem? Please describe.**
So I have tried to use Strawberryshake as GraphQL client in Unity project. I have followed [this](https://chillicream.com/docs/strawberryshake/get-started/console) tutorial to setup schema downloading and code generation. The CLI tool actually works fine and downloads schema + initializes code generation config file. But I have issues with code generation itself. In [this thread](https://forum.unity.com/threads/source-generators-via-roslyn-analyzers-in-2020-2.1003386/) someone mentioned that Unity supports lower Roslyn version, which doesn't support code generation. In fact this feature was supposed to be added in Unity 2021.2, but I can't find anything about it in beta release notes.

**Describe the solution you'd like**
[This](https://github.com/ChilliCream/hotchocolate/issues/1011) issue describes plans for GraphQL + CLI tool. You mention `dotnet graphql compile` command which was supposed to generate C# code based on schema and queries. But as far as I understand it was never implemented. Ability to trigger code generation this way would be really helpful for me.

**Describe alternatives you've considered**
I could have separate dotnet project and generate all code there, but this solution would be annoying. When adding / modifying queries, I would have to do it in separate location and then transfer all files manually.

**Comment by Cheesebaron** [comment]
Can't you already do this. StrawberryShake emits code that is compliant with .NET Standard 2.0. This should be easily consumable from Unity?

**Comment by GMyjak** [comment]
Yes, but the problem is code generation itself. This is why I am asking about CLI tool. And if it would be possible to invoke code generation manually.

**Comment by michaelstaib** [comment]
We are revamping the code generator where we abstract the generator into a code generator server that the source generators will use. With this it would also become possible to integrate it into a cli tool. I am moving this to the October release with no guarantees yet.

**Comment by GMyjak** [comment]
Thank you for your help.

Unfortunately I have found another issue related to Unity. I generated sample client dotnet and checked if it works. Then imported it into Unity and after importing all required dlls I still had one compile error. Client uses ReadOnlySpan but it couldn't be found in System.Runtime dll. 

In [this unity forum thread](https://forum.unity.com/threads/unity-future-net-development-status.1092205/page-7) in comment 327 Josh from Unity team explicitly mentions that they don't support some .NET Standard 2.1 features yet (like Span).

@Cheesebaron mentioned that generated code is compliant with .NET Standard 2.0. But [Microsoft docs](https://docs.microsoft.com/en-us/dotnet/api/system.readonlyspan-1?view=netstandard-2.1) say that ReadOnlySpan was added in .NET Standard 2.1. The only information I could find in StrawberryShake docs is that it is compliant with .NET Standard (without mentioning version).

To sum up, could you provide some official information about .NET Standard compatibility? And could you help me find some workaround that doesn't use ReadOnlySpan? As I understand, simple byte array in its place could work fine, but I cannot replace it manually, because generated class implements some interface from StrawberryShake dll.

**Comment by michaelstaib** [comment]
With version 12.4.0-preview we now support tool base generation without the need for source generators... you can switch between those two.

```bash
dotnet graphql generate [--path] [--defaultNamespace] [--persistedQueryDirectory]  
```

**Comment by tslater** [comment]
@GMyjak , did you ever end up getting it working?

**Comment by mrwellmann** [comment]
For anyone coming here.
I've been able to get it running with using a console project for code generation.
The fixes to make it work on iOS and Android are very crude but it is a starting point.
https://github.com/firedrill-gmbh/Strawberryshake_Unity2021.

**Comment by pixelpax** [comment]
What directory should we build the cli csproj in?

**Comment by mrwellmann** [comment]
@pixelpax sorry the repo was accidentally private now it's public and link should work. How ever the binary's in "Assets/Plugins/StrawberryShake/" are not up to date. I will updat it as soon as I find time.

I still have to try this to be able to update to new versions https://github.com/ChilliCream/hotchocolate/issues/5487#issuecomment-1295498309
